lib/SILGen/SILGenExpr.cpp

```diff
@@ -3544,8 +3544,8 @@ namespace {
 
     // If the destination is a tuple, recursively destructure.
     void visitTupleExpr(TupleExpr *E) {
-      assert(E->getType()->is<TupleType>());
-      assert(!E->getType()->isMaterializable() || E->getType()->isVoid());
+      auto *TTy = E->getType()->castTo<TupleType>();
+      assert(TTy && !TTy->hasInOutElement());
```

Comment:
Oh, can E->getType() have an lvalue type here? In that case, I would instead just assert

```
assert(E->getType()->isLValueType() || E->getType()->isVoid());
```